### PR TITLE
HIVE-25957: Fix password based authentication with SAML enabled

### PR DESCRIPTION
**Pull request by hsnusonic**
### What changes were proposed in this pull request?

A new class AuthType is added to support multiple authentication. This class can parse comma separated list of "hive.server2.authentication" and prohibits invalid values in server start-up.

### Why are the changes needed?

Any password based request in HTTP mode can bypass when SAML is set due to the line below.
```
clientUserName = doPasswdAuth(request, HiveAuthConstants.AuthTypes.NONE.toString());
```

### Does this PR introduce _any_ user-facing change?

Yes, user can set "hive.server2.authentication" to comma separated list now. We currently support SAML with one password base authentication together.

### How was this patch tested?

No new tests added.

### CODE REVIEW

**Review by nrg4878 — COMMENTED** [review]
Looks good to me other than the 1 comment.

Also can we add a UT for the AuthType to test that it accepts "LDAP,SAML" but does not accept "KERBEROS,LDAP,SAML" or any combination of these?

File: service/src/java/org/apache/hive/service/cli/thrift/ThriftHttpServlet.java [hunk 1]
```diff
@@ -551,7 +551,7 @@ String doKerberosAuth(HttpServletRequest request)
       throws HttpAuthenticationException {
     // Each http request must have an Authorization header
     // Check before trying to do kerberos authentication twice
-    getAuthHeader(request, authType);
+    getAuthHeader(request, "NEGOTIATE");
```
Comment (nrg4878) [inline service/src/java/org/apache/hive/service/cli/thrift/ThriftHttpServlet.java:554]: I meant to ask about this earlier. For kerberos auth, if the header contained "NEGOTIATE", how did this work earlier when we were looking for header named "KERBEROS" ?

Comment (hsnusonic) [reply service/src/java/org/apache/hive/service/cli/thrift/ThriftHttpServlet.java]: This argument is unused, so it should be safe to remove it entirely.

Comment (nrg4878) [reply service/src/java/org/apache/hive/service/cli/thrift/ThriftHttpServlet.java]: makes sense. Please mark this comment as resolved.

**Review by nrg4878 — APPROVED** [review]
Changes look good to me. +1

**Pull request merged**